File: endpoints/src/main/java/com/cromoteca/samples/endpoints/FluxEndpoint.java
package com.cromoteca.samples.endpoints;

import com.cromoteca.samples.annotations.Nullable;
import com.vaadin.flow.server.auth.AnonymousAllowed;
import com.vaadin.hilla.Endpoint;
import com.vaadin.hilla.EndpointSubscription;
import java.util.List;
import java.util.Map;
import reactor.core.publisher.Flux;

@Endpoint
@AnonymousAllowed
public class FluxEndpoint {
  public Flux<Map<String, List<@Nullable String>>> flux(List<@Nullable String> arg) {
    return Flux.empty();
  }

  public EndpointSubscription<String> subscription() {
    return EndpointSubscription.of(Flux.empty(), () -> {});
  }
}

File: generator/src/test/resources/com/cromoteca/samples/endpoints/FluxEndpoint.ts
import client from '../../../../connect-client.default.js';
import { type Subscription } from '@hilla/frontend';

function flux(arg: Array<string | undefined>): Subscription<Map<string, Array<string | undefined>>> {
    return client.subscribe('com.cromoteca.samples.endpoints.FluxEndpoint', 'flux', { arg });
}

function subscription(): Subscription<string> {
    return client.subscribe('com.cromoteca.samples.endpoints.FluxEndpoint', 'subscription', {});
}

const FluxEndpoint = {
    flux,
    subscription,
};

export default FluxEndpoint;
